app/commands/exercise/representation/create_or_update.rb

```diff
@@ -6,13 +6,26 @@ class CreateOrUpdate
       initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
 
       def call
+        # First cache the old representation
+        # TODO: This needs tests adding
+        old_source_submission_representation = Exercise::Representation.find_by(source_submission: submission)
+
         representation = Exercise::Representation.find_create_or_find_by!(exercise:, ast_digest:) do |rep|
           rep.source_submission = submission
           rep.ast = ast
           rep.mapping = mapping
           rep.last_submitted_at = last_submitted_at
         end
 
+        # TODO: This needs tests adding
+        if old_source_submission_representation&.has_feedback? && !representation.has_feedback?
+          representation.update!(
+            feedback_author: old_source_submission_representation.feedback_author,
+            feedback_markdown: old_source_submission_representation.feedback_markdown,
+            feedback_type: old_source_submission_representation.feedback_type
+          )
+        end
+
```

Comment:
Small note, that this method clearly needs refactoring, but I don't care for now 🙂